libraries/rush-lib/src/logic/operations/CacheableOperationPlugin.ts

```diff
@@ -0,0 +1,875 @@
+          });
+
+          // Try to acquire the cobuild lock
+          let cobuildLock: CobuildLock | undefined;
+          if (cobuildConfiguration?.cobuildEnabled) {
+            if (
+              cobuildConfiguration?.cobuildLeafProjectLogOnlyAllowed &&
+              rushProject.consumingProjects.size === 0 &&
+              !projectBuildCache
+            ) {
+              // When the leaf project log only is allowed and the leaf project is build cache "disabled", try to get
+              // a log files only project build cache
+              projectBuildCache = await this._tryGetLogOnlyProjectBuildCacheAsync({
+                buildCacheConfiguration,
+                cobuildConfiguration,
+                runner,
+                rushProject,
+                phase,
+                projectChangeAnalyzer,
+                commandName,
+                commandToRun,
+                terminal: buildCacheTerminal,
+                trackedProjectFiles,
+                operationMetadataManager: context._operationMetadataManager
+              });
+              if (projectBuildCache) {
+                buildCacheTerminal.writeVerboseLine(
+                  `Log files only build cache is enabled for the project "${rushProject.packageName}" because the cobuild leaf project log only is allowed`
+                );
+              } else {
+                buildCacheTerminal.writeWarningLine(
+                  `Failed to get log files only build cache for the project "${rushProject.packageName}"`
+                );
+              }
+            }
+
+            cobuildLock = await this._tryGetCobuildLockAsync({
+              runner,
+              projectBuildCache,
+              cobuildConfiguration,
+              packageName: rushProject.packageName,
+              phaseName: phase.name
+            });
+          }
+
+          // eslint-disable-next-line require-atomic-updates -- we are mutating the build cache context intentionally
+          buildCacheContext.cobuildLock = cobuildLock;
+
+          // If possible, we want to skip this operation -- either by restoring it from the
+          // cache, if caching is enabled, or determining that the project
+          // is unchanged (using the older incremental execution logic). These two approaches,
+          // "caching" and "skipping", are incompatible, so only one applies.
+          //
+          // Note that "caching" and "skipping" take two different approaches
+          // to tracking dependents:
+          //
+          //   - For caching, "isCacheReadAllowed" is set if a project supports
+          //     incremental builds, and determining whether this project or a dependent
+          //     has changed happens inside the hashing logic.
+          //
+          //   - For skipping, "isSkipAllowed" is set to true initially, and during
+          //     the process of running dependents, it will be changed by OperationExecutionManager to
```

Comment:
This plugin is the entity that changes isSkipAllowed now, so update the comment.